### PR TITLE
Fix: remove unused connection symmetries and other code cleanups

### PR DESCRIPTION
**Pull request by moubctez**
Changes:
* remove unused connection symmetries (fixes memory draining)
* remove unnecessary cloning
* don't derive DataSize when all fields are skipped
* hide some code that is needed only for tests
* code cleanup

### CODE REVIEW

**Comment by cspramit** [comment]
@moubctez this PR will need review and approval before merge

File: node/src/components/network.rs [hunk 1]
```diff
@@ -203,26 +212,23 @@ where
...
+    #[cfg(test)]
     server_join_handle: Option<JoinHandle<()>>,
 
     /// Channel signaling a shutdown of the incoming connections.
     // Note: This channel is closed when we finished syncing, so the `Network` can close all
     // connections. When they are re-established, the proper value of the now updated `is_syncing`
     // flag will be exchanged on handshake.
-    #[data_size(skip)]
-    close_incoming_sender: Option<watch::Sender<()>>,
+    _close_incoming_sender: Option<watch::Sender<()>>,
```
Comment (EdHastingsCasperAssociation) [inline node/src/components/network.rs]: why retain the field if it is now unused?

Comment (moubctez) [reply node/src/components/network.rs]: It is used for testing.

**Comment by moubctez** [comment]
@EdHastingsCasperAssociation I've sync this PR with feat-2.0 changes, so it's ready for a review.

**Review by EdHastingsCasperAssociation — APPROVED** [review]

**Comment by devendran-m** [comment]
Hi @moubctez , the nctl test is failing. Could you please check it.

**Comment by moubctez** [comment]
I am not able to see the details: Permission denied.

**Comment by devendran-m** [comment]
Hi @moubctez , 
Just to let you know, we shared the error log yesterday.

**Comment by EdHastingsCasperAssociation** [comment]
bors try

**Comment by moubctez** [comment]
@EdHastingsCasperAssociation now it should pass the nctl_upgrade

**Comment by moubctez** [comment]
bors try

**Pull request merged**